gitlab/client.py

```diff
@@ -1,34 +1,36 @@
 """Wrapper for the GitLab API."""
-
 import os
 import re
-import time
+import sys
 from typing import Any, cast, Dict, List, Optional, Tuple, TYPE_CHECKING, Union
-from urllib import parse
 
 import requests
 import requests.utils
-from requests_toolbelt.multipart.encoder import MultipartEncoder  # type: ignore
 
 import gitlab
 import gitlab.config
 import gitlab.const
 import gitlab.exceptions
 from gitlab import utils
 
-REDIRECT_MSG = (
-    "python-gitlab detected a {status_code} ({reason!r}) redirection. You must update "
-    "your GitLab URL to the correct URL to avoid issues. The redirection was from: "
-    "{source!r} to {target!r}"
-)
+from .clients._requestsclient import _RequestsClient
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
 
-RETRYABLE_TRANSIENT_ERROR_CODES = [500, 502, 503, 504] + list(range(520, 531))
 
 # https://docs.gitlab.com/ee/api/#offset-based-pagination
 _PAGINATION_URL = (
     f"https://python-gitlab.readthedocs.io/en/v{gitlab.__version__}/"
     f"api-usage.html#pagination"
 )
+REDIRECT_MSG = (
```

Comment:
Maybe this could use the one in the requestsclient.py file or vice-versa.

Or move it to the constants file. Basically only have one copy of it, if possible.